nixos/modules/services/networking/wg-quick.nix

```diff
@@ -357,8 +374,15 @@ in {
   ###### implementation
 
   config = mkIf (cfg.interfaces != {}) {
-    boot.extraModulePackages = optional (versionOlder kernel.kernel.version "5.6") kernel.wireguard;
-    environment.systemPackages = [ pkgs.wireguard-tools ];
+    boot.extraModulePackages =
+      optional (
+        any (x: x.type == "wireguard") (attrValues cfg.interfaces)
+        && (versionOlder kernel.kernel.version "5.6")
```

Comment:
I think it would be better to check the kernel version first, presuming possible short-circuit evaluation.

Reply:
Nevermind.